app/src/main/java/com.example.weatherapp/mvp/view/WeatherView.kt

```diff
@@ -1,8 +1,9 @@
 package com.example.weatherapp.mvp.view
 
 import android.app.Activity
+import com.example.weatherapp.databinding.ActivityMainBinding
 import com.example.weatherapp.mvp.contract.WeatherContract
 import com.example.weatherapp.mvp.view.base.ActivityView
 
-class WeatherView(activity: Activity): ActivityView(activity), WeatherContract.View {
+class WeatherView(activity: Activity, private val binding: ActivityMainBinding): ActivityView(activity), WeatherContract.View {
```

Comment:
Is this view going to be able to accept every other Activity ? or is it meant to be just used with the main activity, if thats the case we can change Activity for MainActivity

Reply:
I changed the activity that is associate to the view, now only accept one kind of activity, so now activity: WeatherActivity